packages/calypso-e2e/src/lib/components/editor-publish-panel-component.ts

```diff
@@ -0,0 +1,90 @@
+import { Page, Frame, FrameLocator } from 'playwright';
+
+// Options for Playwright's Frame.click() method.
+type ClickOptions = Parameters< Frame[ 'click' ] >[ 1 ];
+type PrePublishButtons = 'Publish' | 'Schedule' | 'Cancel';
+
+const panel = 'div.editor-post-publish-panel';
+const selectors = {
+	// Pre-publish
+	prePublishButton: ( buttonText: PrePublishButtons ) =>
+		`${ panel } .editor-post-publish-panel__header button:text("${ buttonText }")`,
+
+	// Post-publish
+	postPublishClosePanelButton: `${ panel } button[type="button"]:has(svg[aria-hidden="true"])`, // aria-label changes depending on the UI language used.
+	publishedArticleURL: `${ panel } input[readonly]`,
+	addNewArticleButton: `${ panel } .post-publish-panel__postpublish-buttons > a`,
+};
+
+/**
+ * Represents an instance of the WordPress.com Editor's publish checklist & post-publish panel.
+ */
+export class EditorPublishPanelComponent {
+	private page: Page;
+	private frameLocator: FrameLocator;
```

Comment:
Note the use of `FrameLocator`. This is a [new API](https://playwright.dev/docs/1.18/api/class-framelocator) introduced in Playwright 1.17.

It is the iFrame specific counterpart to the `Locator` API and allows us to locate and interact with elements within an iFrame as if we have the actual frame itself.

Previously:
```
const frameHandle = await this.page.waitForSelector('iframe');
const frame = await frameHandle.contentFrame() as Frame;
await frame.click('selector');
```

Now:
```
const locator = this.frameLocator.locator('selector');
await locator.click()
```

Reply:
Great idea!!